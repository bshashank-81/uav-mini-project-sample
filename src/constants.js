export const USERNAME_LABEL = "Username";
export const USERNAME_PLACEHOLDER = "Enter username";
export const PASSWORD_LABEL = "Password";
export const PASSWORD_PLACEHOLDER = "Enter password";
export const SIGN_IN = "Sign in";
export const INVALID_CREDENTIALS = "invalid credentials";
export const NETWORK_ERROR = "network error";
export const LOGIN = "Login";
export const MAPS = "Maps";
export const LOCATIONS = "Locations";
export const LATITUDE = "LATITUDE";
export const LONGITUDE = "LONGITUDE";
export const ID = "ID";
export const NEW_LOCATION = "Add Location";
export const LOGOUT = "Logout";
