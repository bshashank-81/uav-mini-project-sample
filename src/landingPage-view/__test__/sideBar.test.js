import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { create } from "react-test-renderer";
import SideBar from "../components/sideBar";
import { shallow, mount } from "enzyme";
import { fireEvent, render } from "@testing-library/react";
import * as Constants from "../../constants";
import { Link } from "react-router-dom";
import Drawer from "@material-ui/core/Drawer";
import { Typography, Box } from "@material-ui/core";
import { MemoryRouter } from "react-router-dom";

describe("sidebar component", () => {
  const logoutUser = jest.fn();
  test("Matches the snapshot", () => {
    const sidebar = create(
      <Router>
        <SideBar />
      </Router>
    );
    expect(sidebar.toJSON()).toMatchSnapshot();
  });
});
