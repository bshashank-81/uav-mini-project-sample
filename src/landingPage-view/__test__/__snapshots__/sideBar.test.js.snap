// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sidebar component Matches the snapshot 1`] = `
<div>
  <div
    className="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-1"
  >
    <div
      className="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-2 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
    >
      <hr
        className="MuiDivider-root"
      />
      <div>
        <a
          className="makeStyles-links-4"
          href="/login"
          onClick={[Function]}
        >
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root makeStyles-listButtons-3 MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiTypography-root MuiTypography-body1"
              data-testid="login"
            >
              <div
                className="MuiBox-root MuiBox-root-6"
              >
                Login
              </div>
            </div>
          </div>
        </a>
        <a
          className="makeStyles-links-4"
          href="/maps"
          onClick={[Function]}
        >
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root makeStyles-listButtons-3 MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiTypography-root MuiTypography-body1"
              data-testid="maps"
            >
              <div
                className="MuiBox-root MuiBox-root-7"
              >
                Maps
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
